dashboard/src/actions/toastActions.js

```diff
@@ -21,15 +19,33 @@ export const showFailureToast = () => async (dispatch) => {
     title: "Something went wrong",
     message: "Please try again later",
   };
-  dispatch({
-    type: TYPES.SHOW_TOAST,
-    payload: toast,
-  });
+  dispatch(showToast(toast.variant, toast.title, toast.message));
 };
 
-export const constructToast = (variant, title, message = "") => {
-  return {
-    type: TYPES.SHOW_TOAST,
-    payload: { variant, title, message },
+export const showToast =
+  (variant, title, message = "") =>
+  (dispatch, getState) => {
+    const alerts = getState().toastReducer.alerts;
+    const obj = {
+      variant: variant,
+      title: title,
+      message: message,
+      key: uid(),
+    };
+    alerts.push(obj);
+
+    dispatch({
+      type: TYPES.SHOW_TOAST,
+      payload: alerts,
+    });
```

Comment:
As I understand it, `alerts` is a reference to part of the Redux state, and we're not supposed to mutate that; yet, line 35 is doing so.  It's true that the mutation doesn't last for long before we update the state, but I think we're still technically outside the rules.

So, the code should probably do something more like this:
```
    const obj = {
      variant: variant,
      title: title,
      message: message,
      key: uid(),
    };
    const alerts = [...getState().toastReducer.alerts, obj]

    dispatch({
      type: TYPES.SHOW_TOAST,
      payload: alerts,
    });
```
which creates the object that we want to add to the alerts list, then creates a copy of the alerts list with the object added at the end (leaving the Redux state unchanged), and then passes the copy to `dispatch()` to be used in the update of the state.